### PR TITLE
[GH#48077]: Updating etcd manual defragmentation docs

### PR DESCRIPTION
**Pull request by lahinson**
[GH#48077]: Updating etcd manual defragmentation docs

Version(s): 4.10, 4.11, 4.12

Issue: https://github.com/openshift/openshift-docs/issues/48077

Link to docs preview (VPN required): http://file.rdu.redhat.com/lahinson/etcd-manual-defrag/post_installation_configuration/cluster-tasks.html#manual-defrag-etcd-data_post-install-cluster-tasks

Additional information:

### CODE REVIEW

**Comment by lahinson** [comment]
@dusk125 When you can, would you take a look at the updated docs for manually defragmenting etcd and let me know what you think? I based the updates on our Slack conversation. See the preview link in the first comment 👆 Thanks!

**Comment by dusk125** [comment]
Yeah looks good to me! I don't know if it would be appropriate to include the alert names? I'm not sure how in-depth this doc should go.

**Comment by lahinson** [comment]
@dusk125 Thanks! I can run the draft past the initial requester and see whether he might need more detailed info. Now that I'm looking at the text again, I have more question for you. Here's the text:

>If etcd uses more than 50% of its available space, a Prometheus alert indicates that you need to use manual defragmentation. You also need to use manual defragmentation if etcd uses close to the predefined quota for the instance.

Are those two sentences kind of repetitive? I'm wondering whether using "close to the predefined quota" is similar to using "more than 50% of its available space". Or is the predefined quota something different than the amount of available space?

**Comment by dusk125** [comment]
@lahinson Maybe could rephrase to something like:
```
If etcd uses more than 50% of its available space or it's fragmentation ratio is high (size in use / total database size), a Prometheus alert indicates that you may need to use manual defragmentation.
After defragmentation, if etcd is still using more than 50% of its available space, consider increasing the disk quota for etcd.
```

**Comment by dusk125** [comment]
Sorry, should probably also have the threshold for the fragmentation ratio:
`size in use / total database size < 0.5`. In other words, etcd is only actively using less than half of the actual space it's taking up.

**Comment by lahinson** [comment]
@dusk125 Will the phrase "fragmentation ratio" be meaningful to the target audience, or do you think it might be clearer to say this: "If etcd uses more than 50% of its available space or it is using less than half of its total database size, a Prometheus alert indicates that you might need to use manual defragmentation."

**Comment by dusk125** [comment]
@lahinson You're right, probably better to avoid the phrase and use the latter for clarity.

A side note, I'm not sure if this is implied or common knowledge or worth including, but the alert doesn't fire when the threshold is met instantaneously, it will fire when etcd has sat at the threshold (either 50% or high ratio) for 10 minutes. I just realized that reading through my suggestion again, one could interpret it as firing as soon as the threshold was met (could very well be an incorrect interpretation for the audience).

**Comment by lahinson** [comment]
@dusk125 Thanks for the clarification! 

I tried to fit in the part about waiting about 10 minutes, but the sentences were getting a little unwieldy, so I went a different route. Does this look correct?

>A Prometheus alert indicates when you need to use manual defragmentation. The alert is displayed in two cases:
>
  > * When etcd uses more than 50% of its available space for more than 10 minutes
   >* When etcd uses less than 50% of its total database size for more than 10 minutes

**Comment by dusk125** [comment]
@lahinson Looks good! I think the only thing I would suggest is for the second statement emphasizing the difference between it taking up more than 50% of its quota (first statement) and how much space its actively using.

So maybe the second something like: 
> When etcd actively uses less than 50% of its total database size for more than 10 minute?

**Comment by lahinson** [comment]
Thanks again @dusk125! Here's the latest update (VPN required): http://file.rdu.redhat.com/lahinson/etcd-manual-defrag/post_installation_configuration/cluster-tasks.html#etcd-defrag_post-install-cluster-tasks

Note: I also created a "Next steps" section at the end of the defrag procedure and moved the line about considering increasing the disk quota there.

**Comment by dusk125** [comment]
Anytime! Happy to help!

**Comment by lahinson** [comment]
Hi @geliu2016 - When you can, please take a look at this update to the manual defragmentation docs and let me know if you have any comments. Thanks!

File: modules/etcd-defrag.adoc [hunk 1]
```diff
@@ -51,11 +51,16 @@ failed defrag on member: __<member_name>__, memberID: __<member_id>__: __<error_
 [id="manual-defrag-etcd-data_{context}"]
 == Manual defragmentation
 
-You can monitor the `etcd_db_total_size_in_bytes` metric to determine whether manual defragmentation is necessary.
+//You can monitor the `etcd_db_total_size_in_bytes` metric to determine whether manual defragmentation is necessary.
+
+A Prometheus alert indicates when you need to use manual defragmentation. The alert is displayed in two cases:
+
+   * When etcd uses more than 50% of its available space for more than 10 minutes
+   * When etcd is actively using less than 50% of its total database size for more than 10 minutes
```
Comment (stevsmit) [inline modules/etcd-defrag.adoc:59]: Should we have periods at the end of these?

Comment (lahinson) [reply modules/etcd-defrag.adoc:59]: @stevsmit If I'm reading IBM Style [1] correctly, I don't think so. Both list items are sentence fragments, not complete sentences. 

[1] https://www.ibm.com/docs/en/ibm-style?topic=format-lists#punctuation-in-lists

Comment (stevsmit) [reply modules/etcd-defrag.adoc:59]: I figured it was because they were incomplete sentences. Thanks!

**Pull request merged**